### PR TITLE
Removing headlessui used in menus

### LINKED ISSUE #2073: replace headlessui with radix-ui
we still have some places where we use headless-ui
image

instead of `radix-ui`. Lets replace those components and remove the dependency.

`import { Switch } from "@headlessui/react";` => https://www.radix-ui.com/docs/primitives/components/switch

`import { Menu, Transition } from "@headlessui/react";` => https://www.radix-ui.com/docs/primitives/components/dropdown-menu

We need a new `<Menu/>` component (similar to `components/ui/Dropdown/Dropdown.tsx`) 

remember to abide by the tailwind design system and not use the default radix-ui styles

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Recreating PR #2114, now it does not depend on any other branch.

Fixes #2073 

## Type of change

- [X] Technical debt

## How should this be tested?

 - Go to Event type page in your mobile phone and see the mobile-specific dropdown to be Radix and have UI/UX consistency to the rest of menus
- Go to Bookings page in your mobile phone and see the mobile-specific dropdown for a booking to be Radix and have UI/UX consistency to the rest of menus

### CODE REVIEW

File: apps/web/components/team/TeamListItem.tsx [hunk 1]
```diff
@@ -99,7 +99,7 @@ export default function TeamListItem(props: Props) {
             </>
           )}
           {!isInvitee && (
-            <div className="flex space-x-2 rtl:space-x-reverse">
```
Comment (leog) [inline apps/web/components/team/TeamListItem.tsx:102]: When the dropdown gets included in the same context (when not "portalled"), there is an extra space added by this util, so I removed it. Spacing does not look bad without it.

File: apps/web/components/ui/Dropdown.tsx [hunk 2]
```diff
@@ -25,8 +25,11 @@ export const DropdownMenuContent = forwardRef<HTMLDivElement, DropdownMenuConten
   ({ children, ...props }, forwardedRef) => {
     return (
       <DropdownMenuPrimitive.Content
+        portalled={false}
```
Comment (leog) [inline apps/web/components/ui/Dropdown.tsx:28]: As explained, this makes the dropdown be added in the `body` tag when "portalled".

File: apps/web/components/ui/Dropdown.tsx [hunk 3]
```diff
@@ -25,8 +25,11 @@ export const DropdownMenuContent = forwardRef<HTMLDivElement, DropdownMenuConten
...
         {...props}
-        className="z-10 mt-1 w-48 origin-top-right rounded-sm bg-white text-sm shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none"
+        className={`${
+          props.portalled ? `` : `md:-ml-[55px]`
+        } z-10 mt-1 -ml-0 w-48 origin-top-right rounded-sm bg-white text-sm shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none lg:-ml-56`}
```
Comment (leog) [inline apps/web/components/ui/Dropdown.tsx:32]: As there a quite a few views with different toolsets, i.e. full sidebar, collapsed sidebar and no sidebar (bottom bar), I needed to take that into consideration. These tweaks are not needed when the dropdown is "portalled".

**Review by PeerRich — APPROVED** [review]
great stuff

File: apps/web/components/ui/Dropdown.tsx (hunk 2 above)
Comment (zomars) [reply apps/web/components/ui/Dropdown.tsx:28]: I'm not 100% sold on the nomenclature. Since is a boolean this could be `isPortal` to improve legibility.

**Pull request merged**